Tools/meta/bug fixes

ehildenb opened the pull request.

ehildenb commented on contrib/tools/meta.md/cterms.md:
```diff
@@ -203,15 +203,15 @@ Right now we just take *any* maximal sort in `C1 /\ C2`, but we really need to t
-    eq joint-sort(S, ME,  ME')  = joint-sort(S, intersect(ME, ME'), intersect(ME, ME')) .
+    eq joint-sort(S, ME,  ME')  = joint-sort(S, moduleIntersect(ME, ME'), moduleIntersect(ME, ME')) .
     eq joint-sort(S, MOD, MOD') = joint-sort(S, asTemplate(MOD), asTemplate(MOD')) .
 
```
This looks strange to me, why would `joint-sort` effectively pass the same module as both arguments here? It calls `moduleIntersect(ME, ME')` in both positions. This bug goes back to when the code was introduced, I think it should be `joint-sort(S, upModule(ME, true), upModule(ME', true))` instead.

nishantjr replied:
Good catch. I'll cook up a test case before fixing it.

Commit:
meta/cterms: joint-sort on module-expressions reduces to joint-sort on modules

nishantjr committed:
contrib/.../bitswap: Begin modeling as game with players taking turns at each round of the protocol

nishantjr committed:
contrib/.../bitswap: Implement strategies "naive" and "round-robin"

nishantjr committed:
contrib/.../bitswap: Documentation

nishantjr committed:
contrib/.../bitswap: Change equations to rewrites to allow more non-determinism

nishantjr committed:
contrib/.../bitswap: Create a narrative and outline for what needs to be implemented

nishantjr committed:
contrib/Makefile, bitswap: tangle blocks marked `test` into a file in `test/`

nishantjr committed:
contrib/.../bitswap: Tweak rules for easier printing of transfers of blocks

nishantjr committed:
contib/.../bitswap: Get list of IDs before executing strategies over WorldState

This means that rewrites do not happen inside `get-id` and hugely reduces the
search space. It also removes redundant prints.

nishantjr committed:
contrib/.../bitswap: Cleanup, references and more text

nishantjr committed:
contrib/.../bitswap: Minor formatting and Jose's corrections

nishantjr committed:
contrib/.../bitswap: tests (and demos) of the naive and round-robin strategies

nishantjr committed:
contrib/.../bitswap: Use `crl` and matching to improve readability

nishantjr committed:
contrib/.../bitswap: formatting

nishantjr committed:
contrib/.../bitswap: More test for round-robin

nishantjr committed:
contrib/.../bitswap: tick: takes a Nat, specifying number of rounds

nishantjr committed:
contrib/.../bitswap: Implement strategy `tit-for-tat`

nishantjr committed:
contrib/Makefile: recipe for contrib/systems.md/bitswap.pdf

Commit:
contrib/systems/bitswap: CS 476 project (#8: systems/bitswap)
